plan/physical_plans.go

```diff
@@ -334,6 +338,11 @@ type PhysicalUnionScan struct {
 	Conditions []expression.Expression
 }
 
+// IsPartition returns true and partition ID if it works on a partition.
+func (p *PhysicalIndexScan) IsPartition() (bool, int64) {
```

Comment:
How about s/IsPartition/GetPartitionID/

Reply:
I think `IsPartition` maybe better.

```
succ, _ :=  IsPartition()
if succ {
    doSomething()
}
```

It's also flexible when we need the partition id:

```
if succ, pid := IsPartition(); succ {
    doSomthingWith(pid)
}
```

`GetPartitionID` looks unnatural if we define:

```
GetPartitionID(pid int64, bool) // wow, it maybe not a partition
```

If we use `partition id == 0` to represent it's not a partition:

```
GetPartitionID(pid int64)
```

This code introduce an implicit assumption with the caller.